code/modules/hydroponics/spreading/spreading.dm

```diff
@@ -0,0 +1,195 @@
+#define DEFAULT_SEED "glowshroom"
+#define CREEPER_GROWTH_DISTANCE 4
+
+/obj/effect/plantsegment
+	name = "space vines"
+	desc = "An extremely expansionistic species of vine."
+	icon = 'icons/effects/spacevines.dmi'
+	icon_state = "Light1"
+	anchored = 1
+	opacity = 0
+	density = 0
+	layer = 5
+	pass_flags = PASSTABLE | PASSGRILLE
+	//flags = PROXMOVE
+	//mouse_opacity = 2
+
+	var/health = 10
+	var/max_health = 100
+	var/list/neighbors = list()
+	var/turf/epicenter
+	var/datum/seed/seed
+	var/sampled = 0
+	var/spread_chance
+	var/spread_distance
+	var/mature_time
+	var/last_tick = 0
+	var/harvest = 0
+	var/age = 0
+	var/limited_growth = 0
+	var/plant_damage_noun = "Thorns"
+
+/obj/effect/plantsegment/creeper
+	limited_growth = 1
+/obj/effect/plantsegment/single
+	spread_chance = 0
+
+/obj/effect/plantsegment/Destroy()
+	if(plant_controller)
+		plant_controller.remove_plant(src)
+	for(var/obj/effect/plantsegment/neighbor in range(1,src))
+		plant_controller.add_plant(neighbor)
+	..()
+
+/obj/effect/plantsegment/New(var/newloc, var/datum/seed/newseed, var/turf/newepicenter, var/start_fully_mature = 0)
+	..()
+
+	if(!newepicenter)
+		epicenter = get_turf(src)
+	else
+		epicenter = newepicenter
+
+	if(!plant_controller)
+		sleep(250) // ugly hack, should mean roundstart plants are fine.
+	if(!plant_controller)
+		world << "<span class='danger'>Plant controller does not exist and [src] requires it. Aborting.</span>" //This probably shouldn't be a world <<
+		qdel(src)
+		return
+
+	if(!istype(newseed))
+		newseed = plant_controller.seeds[DEFAULT_SEED]
+	seed = newseed
+	if(!seed)
+		qdel(src)
+		return
+
+	name = "[seed.seed_name] vines"
+	max_health = round(seed.endurance/2)
+	if(seed.spread == 1)
+		limited_growth = 1
+	mature_time = Ceiling(seed.maturation/2)
+	spread_chance = round(40 + triangular_seq(seed.potency*2, 30)) // Diminishing returns formula, see maths.dm
+	spread_distance = limited_growth ? (CREEPER_GROWTH_DISTANCE) : round(spread_chance*0.2)
+	update_icon()
+
+	if(start_fully_mature)
+		health = max_health
+		mature_time = 0
+
+	spawn(1) // Plants will sometimes be spawned in the turf adjacent to the one they need to end up in, for the sake of correct dir/etc being set.
+		plant_controller.add_plant(src)
+		// Some plants eat through plating.
+		if(seed.chems && !isnull(seed.chems["pacid"]))
+			var/turf/T = get_turf(src)
+			T.ex_act(prob(80) ? 3 : 2)
+
+/obj/effect/plantsegment/examine(mob/user)
+	..()
+	if(!seed) return
+	var/traits = ""
+	if(seed.stinging) traits += "<span class='alert'>It's covered in tiny stingers.</span> "
+	if(seed.thorny) traits += "<span class='alert'>It's covered in sharp thorns.</span> "
+	if(seed.ligneous) traits += "It's a tough and hard vine that can't be easily cut. "
+	if(seed.carnivorous == 2) traits += "It's quivering viciously. "
+	if(seed.hematophage) traits += "It's roots are blood red... "
+	if(traits) user << traits
```

Comment:
Reading all these things on a seperate line is much nicer than on 1, personally.